src/iceflow/data/fetch.py

```diff
@@ -11,7 +12,7 @@ def search_and_download(
     version: str,
     short_name: str,
     bounding_box: Sequence[float],
-    temporal: tuple[str, str],
+    temporal: tuple[dt.datetime | dt.date, dt.datetime | dt.date],
```

Comment:
You can technically do `tuple[dt.date, dt.date]`, as `dt.datetimes` are `dt.dates` too:

```python
>>> import datetime as dt
>>> date = dt.datetime.today().date()
>>> isinstance(date, dt.datetime)
False
>>> isinstance(date, dt.date)
True
>>> datetime = dt.datetime.today()
>>> isinstance(datetime, dt.datetime)
True
>>> isinstance(datetime, dt.date)
True
```